frontend/lib/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -303,27 +299,58 @@ function PlotlyFigure({
     )
   }
 
-  const { data, layout, frames } = spec
-
   const reset = useCallback((): void => {
-    widgetMgr.setElementState(element.id, SELECTIONS, {})
-    widgetMgr.setElementState(element.id, DATA, {})
     widgetMgr.setStringValue(element, "{}", { fromUi: true }, fragmentId)
-  }, [widgetMgr, element, fragmentId])
+
+    // Reset the selection info within the plotly figure
+    setPlotlyFigure({
+      ...plotlyFigure,
+      data: plotlyFigure.data.map((trace: any) => {
+        return {
+          ...trace,
+          // Set to null to clear the selection an empty
+          // array here would still show everything as opaque
+          selectedpoints: null,
+        }
+      }),
+      layout: {
+        ...plotlyFigure.layout,
+        selections: [],
+      },
+    })
+  }, [plotlyFigure, widgetMgr, element, fragmentId])
+
+  // This is required for the form clearing functionality:
+  React.useEffect(() => {
+    const formClearHelper = new FormClearHelper()
+    formClearHelper.manageFormClearListener(widgetMgr, element.formId, reset)
+
+    return () => {
+      formClearHelper.disconnect()
+    }
+  }, [element.formId, reset, widgetMgr])
 
   return (
     <Plot
       key={isFullScreen ? "fullscreen" : "original"}
       className="stPlotlyChart"
-      data={data}
-      layout={layout}
-      config={config}
-      frames={frames}
+      data={plotlyFigure.data}
+      layout={plotlyFigure.layout}
+      config={plotlyConfig}
+      frames={plotlyFigure.frames}
       onSelected={
         element.isSelectEnabled || !disabled ? handleSelect : () => {}
       }
       onDoubleClick={element.isSelectEnabled || !disabled ? reset : () => {}}
       onDeselect={element.isSelectEnabled || !disabled ? reset : () => {}}
+      onInitialized={figure => {
+        widgetMgr.setElementState(element.id, "figure", figure)
+      }}
+      onUpdate={figure => {
+        console.log("onUpdate", figure)
```

Comment:
nit tho this PR is technically a draft... console.log here

Reply:
Removed